ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -280,6 +293,49 @@ - (void)testShouldHandleVnodeOpEvent {
     XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
 }
 
+- (void)testCurrentProcessWasSpawnedByRegularUser {
+    // Defaults should pass for all tests
+    XCTAssertTrue(CurrentProcessWasSpawnedByRegularUser());
+    MockProcess_Reset();
+
+    // Process tree with all UIDs less than 500 were not spawned by regular users
+    MockProcess_AddContext(context, 1 /*pid*/);
+    MockProcess_SetSelfPid(1);
+    MockProcess_AddCredential(1 /*credentialId*/, 1 /*UID*/);
+    MockProcess_AddProcess(1 /*pid*/, 1 /*credentialId*/, 1 /*ppid*/, "test" /*name*/);
+    XCTAssertFalse(CurrentProcessWasSpawnedByRegularUser());
+    MockProcess_Reset();
+
+    // Test a process with a service UID, valid parent pid, but proc_find fails to find parent pid
+    MockCalls::Clear();
+    MockProcess_AddContext(context, 1 /*pid*/);
+    MockProcess_SetSelfPid(1);
+    MockProcess_AddCredential(1 /*credentialId*/, 1 /*UID*/);
+    MockProcess_AddProcess(1 /*pid*/, 1 /*credentialId*/, 501 /*ppid*/, "test" /*name*/);
+    XCTAssertFalse(CurrentProcessWasSpawnedByRegularUser());
+    XCTAssertTrue(MockCalls::DidCallFunction(KextMessageLogged, KEXTLOG_ERROR));
+    MockProcess_Reset();
+
+    // 'sudo' scenario: Root process with non-root parent
+    MockProcess_AddContext(context, 1 /*pid*/);
+    MockProcess_SetSelfPid(1);
```

Comment:
Same comment here:  Consider making the process tree with 1 as the top-most parent:

```
   1 (UID: 1)
   |
  501 (UID: 501)
   |
  502 (self, UID: 1)